File: samples/flask-mongo-blog/apphost.py
import sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).parent / ".modules"))

from aspire import create_builder

builder = create_builder()

mongo = builder.add_mongo_db("mongodb") \
    .with_mongo_express()

db = mongo.add_database("blogdb")

builder.add_python_app("blog", "./src", "main.py") \
    .with_reference(db) \
    .with_http_endpoint(env="PORT") \
    .with_external_http_endpoints()

app = builder.build()
app.run()

File: samples/flask-mongo-blog/apphost.cs
#:package Aspire.Hosting.Python@9.2.1
#:package Aspire.Hosting.MongoDB@9.2.1
#:sdk Aspire.AppHost.Sdk@9.2.1

var builder = DistributedApplication.CreateBuilder(args);

var mongo = builder.AddMongoDB("mongodb")
    .WithMongoExpress();

var db = mongo.AddDatabase("blogdb");

builder.AddPythonApp("blog", "./src", "main.py")
    .WithReference(db)
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints();

builder.Build().Run();
